src/app/shipping/MultiShippingForm.tsx

```diff
@@ -87,32 +103,92 @@ class MultiShippingForm extends PureComponent<MultiShippingFormProps & WithLangu
-                                onSelectAddress={ this.handleSelectAddress }
-                                onUseNewAddress={ onUseNewAddress }
-                            />
-                        </li>
-                    )) }
-                </ul>
-
-                <ShippingFormFooter
-                    cartHasChanged={ cartHasChanged }
+            <Fragment>
+                { !!itemAddingAddress && <AddressFormModal
+                    countries={ countries }
+                    countriesWithAutocomplete={ countriesWithAutocomplete }
+                    defaultCountryCode={ defaultCountryCode }
+                    getFields={ getFields }
+                    googleMapsApiKey={ googleMapsApiKey }
                     isLoading={ isLoading }
-                    isMultiShippingMode={ true }
-                    shouldDisableSubmit={ this.shouldDisableSubmit() }
-                    shouldShowOrderComments={ shouldShowOrderComments }
-                    shouldShowShippingOptions={ !hasUnassignedLineItems(consignments, cart.lineItems) }
-                />
-            </Form>
+                    isOpen={ true }
+                    onRequestClose={ this.handleCloseAddAddressForm }
+                    onSaveAddress={ this.handleSaveAddress }
+                /> }
+
+                <Form>
+                    <ul className="consignmentList">
+                        { items.map(item => (
+                            <li key={ item.key }>
+                                <ItemAddressSelect
+                                    addresses={ addresses }
+                                    item={ item }
+                                    onSelectAddress={ this.handleSelectAddress }
+                                    onUseNewAddress={ this.handleUseNewAddress }
+                                />
+                            </li>
+                        )) }
+                    </ul>
+
+                    <ShippingFormFooter
+                        cartHasChanged={ cartHasChanged }
+                        isLoading={ isLoading }
+                        isMultiShippingMode={ true }
+                        shouldDisableSubmit={ this.shouldDisableSubmit() }
+                        shouldShowOrderComments={ shouldShowOrderComments }
+                        shouldShowShippingOptions={ !hasUnassignedLineItems(consignments, cart.lineItems) }
+                    />
+                </Form>
+            </Fragment>
         );
     }
 
+    private handleSaveAddress: (address: AddressFormValues) => void = async address => {
+        const { createCustomerAddress } = this.props;
+        const { itemAddingAddress } = this.state;
+
+        if (!itemAddingAddress) {
+            return;
+        }
+
+        const shippingAddress = mapAddressFromFormValues(address);
+
+        try {
+            createCustomerAddress(shippingAddress);
+        } catch (_) {
+            // Fail silently if customer address was not saved.
```

Comment:
What do you mean by fail silently? Why don't we show an alert or something to tell the shopper that something went wrong? It feels odd if we don't give any visual feedback and nothing is saved after clicking on the save button?

Reply:
The consignment would have got the address updates, but not saved in the customer address book. The intention was to handle more gracefully the scenario where customer service is down.

Reply:
I didn't think about address creation for other reasons as we are supposed to validate the input in the frontend anyway. Only submit a form when the values are actually valid.

If the service was down, no address would be saved so I thought it could be quite disruptive to show that error continuously for every new address, but I guess it just takes a single click to close a modal.